### PR TITLE
[Merged by Bors] - chore(Topology): golf, move

### PR DESCRIPTION
**Pull request by urkud**
Golf and move `Prod.instNeBotNhdsWithinIio` etc.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Basic.lean [hunk 1]
```diff
@@ -1273,6 +1273,11 @@ lemma nhdsWithin_neBot : (𝓝[s] x).NeBot ↔ ∀ ⦃t⦄, t ∈ 𝓝 x → (t
   exact forall₂_congr fun U _ ↦
     ⟨fun h ↦ h (mem_principal_self _), fun h u hsu ↦ h.mono $ inter_subset_inter_right _ hsu⟩
 
+@[gcongr]
```
Comment (grunweg) [inline Mathlib/Topology/Basic.lean:1276]: For other reviewers: this was just moved (and the gcongr tag is new).

File: Mathlib/Topology/ContinuousOn.lean [hunk 2]
```diff
@@ -325,22 +323,34 @@ theorem nhdsWithin_pi_eq {ι : Type*} {α : ι → Type*} [∀ i, TopologicalSpa
...
 #align nhds_within_pi_eq_bot nhdsWithin_pi_eq_bot
 
-theorem nhdsWithin_pi_neBot {ι : Type*} {α : ι → Type*} [∀ i, TopologicalSpace (α i)] {I : Set ι}
-    {s : ∀ i, Set (α i)} {x : ∀ i, α i} : (𝓝[pi I s] x).NeBot ↔ ∀ i ∈ I, (𝓝[s i] x i).NeBot := by
+theorem nhdsWithin_pi_neBot {I : Set ι} {s : ∀ i, Set (π i)} {x : ∀ i, π i} :
+    (𝓝[pi I s] x).NeBot ↔ ∀ i ∈ I, (𝓝[s i] x i).NeBot := by
   simp [neBot_iff, nhdsWithin_pi_eq_bot]
 #align nhds_within_pi_ne_bot nhdsWithin_pi_neBot
 
+instance instNeBotNhdsWithinUnivPi {s : ∀ i, Set (π i)} {x : ∀ i, π i}
```
Comment (grunweg) [inline Mathlib/Topology/ContinuousOn.lean:341]: This instance is new.

File: Mathlib/Topology/ContinuousOn.lean [hunk 3]
```diff
@@ -325,22 +323,34 @@ theorem nhdsWithin_pi_eq {ι : Type*} {α : ι → Type*} [∀ i, TopologicalSpa
...
+    [∀ i, (𝓝[s i] x i).NeBot] : (𝓝[pi univ s] x).NeBot := by
+  simpa [nhdsWithin_pi_neBot]
+
+instance Pi.instNeBotNhdsWithinIio [Nonempty ι] [∀ i, Preorder (π i)] {x : ∀ i, π i}
```
Comment (grunweg) [inline Mathlib/Topology/ContinuousOn.lean:345]: And these proofs are golfed (nice!).

**Review by grunweg — COMMENTED** [review]
Thank you for doing this - the changes look good to me!
maintainer merge

**Review by jcommelin — COMMENTED** [review]
bors merge

**Pull request closed without merging**